### PR TITLE
Add data.Marshaler

### PR DESCRIPTION
**Pull request by ttacon**
This just adds the interface and a simple test case (the exact issue from issue #18 ).

### CODE REVIEW

File: data/convert.go [hunk 1]
```diff
@@ -103,3 +108,9 @@ func (c StructOptions) Data(obj interface{}) Map {
 	}
 	return Map(m)
 }
+
+// Marshaler is the interface implemented by entities that can marshal
+// themselves into a data.Map.
```
Comment (robfig) [inline data/convert.go]: s/Map/Value/

**Comment by robfig** [comment]
nice, thanks

**Comment by ttacon** [comment]
There's always something that slips my attention :P

**Comment by robfig** [comment]
LGTM :)

**Pull request merged**